src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1653,6 +1654,22 @@ format layout_optimizer::get_preferred_format(program_node& node) {
     auto output_layout = node.get_output_layout();
     bool use_onednn_impls = _optimization_attributes.use_onednn_impls;
 
+    bool allow_new_shape_infer = node.get_program().get_options().get<build_option_type::allow_new_shape_infer>()->enabled();
+
+    if (allow_new_shape_infer) {
+        if (node.is_type<shape_of>())
+            return format::get_default_format(node.get_dependency(0).get_output_layout(false).get_rank());
+        for (auto u : node.get_users()) {
+            for (auto dep_idx : u->get_shape_infer_dependencies()) {
+                if (u->get_dependencies().size() <= dep_idx)
+                    continue;
```

Comment:
Is it okay for this situation to happen? I wonder what it means that `dep_idx` is greater than `get_dependencies().size()`.

Reply:
We are specifying "possible" shape infer dependency in shape_infer dep. However the dep does not always existing. 
For example reshape has dep to idx "1", but it does not have memory dependency to that index, if the pattern memory is given with constant or predefined values, not from the memory.

Reply:
Thank you for your kind explanation!